File: Code/BioMed-DI/IdentityResolution/src/main/java/genes/IdentityResolution/solutions/DI2Datasets.java
package genes.IdentityResolution.solutions;

import java.io.File;

import de.uni_mannheim.informatik.dws.winter.model.HashedDataSet;
import de.uni_mannheim.informatik.dws.winter.model.defaultmodel.Attribute;
import genes.IdentityResolution.model.Gene.Gene;
import genes.IdentityResolution.model.Gene.GeneXMLReader;

public class DI2Datasets {


    public static HashedDataSet<Gene, Attribute> kaessmann() throws Exception {

        HashedDataSet<Gene, Attribute> ds = new HashedDataSet<>();

        new GeneXMLReader().loadFromXML(new File("data/input/DI2/Kaessmann_dt.xml"), "/genes/gene", ds);

        return ds;

    }


    public static HashedDataSet<Gene, Attribute> all_gene_disease_pmid_associations(int i) throws Exception {

        HashedDataSet<Gene, Attribute> ds = new HashedDataSet<>();

        new GeneXMLReader().loadFromXML(new File("data/input/DI2/all_gene_disease_pmid_associations_" + i + "_dt.xml"), "/genes/gene", ds);

        return ds;

    }
 
    
}

File: Code/BioMed-DI/DataFusion2/DI2/DI2Datasets.cs
using System;
using System.IO;
using System.Collections.Generic;

namespace DataFusion2
{
    class DI2Datasets
    {

        public static List<FileInfo> datasets()
        {

            List<FileInfo> di2datasets = new List<FileInfo>();

            di2datasets.Add(Kaessmann);

            for (int i = 1; i <= 7; i++)
            {

                di2datasets.Add(all_gene_disease_pmid_associations(i));
                
            }

            return di2datasets;

        }

        public static FileInfo Kaessmann = new FileInfo(Environment.CurrentDirectory + "/data/input/DI2/Kaessmann_dt.xml");

        public static FileInfo all_gene_disease_pmid_associations(int i) 
        {
            return new FileInfo(Environment.CurrentDirectory + "/data/input/DI2/all_gene_disease_pmid_associations_" + i + "_dt.tsv");
        }

    }

}